tests/vagrant/test_vagrant.py

```diff
@@ -0,0 +1,131 @@
+    assert host.user("vagrant").exists
+    assert host.group("vagrant").exists
+    assert host.user("vagrant").uid == 1000
+    assert host.user("vagrant").gid == 1000
+
+
+def test_vagrant_sudoers_file(host):
+    """Check if vagrant user's sudoers file is present and correct"""
+    with host.sudo():
+        sudoers_file = host.file("/etc/sudoers.d/vagrant")
+        sudoers_file.contains("vagrant     ALL=(ALL)     NOPASSWD: ALL")
+
+
+def test_guest_tools_installed(host):
+    """Check if Hypervisor Guest Additions/Tools/Agents/Kernel modules installed"""
+    kvm_tools = ['qemu-guest-agent', 'rsync', 'nfs-utils']
+    hyperv_tools = ['cifs-utils', 'hyperv-daemons']
+    hypervisor = host.check_output("systemd-detect-virt")
+    if hypervisor == "kvm":
+        for package in kvm_tools:
+            assert host.package(package).is_installed
+    elif hypervisor == "oracle":
+        vb_guest_cmd = host.run("lsmod | grep vboxguest")
+        assert vb_guest_cmd.succeeded == True
+        assert host.package("nfs-utils").is_installed
+    elif hypervisor == "vmware":
+        assert host.package("open-vm-tools").is_installed
+        assert host.package("nfs-utils").is_installed
+    elif hypervisor == "microsoft":
+        for package in hyperv_tools:
+            assert host.package(package).is_installed
+    else:
+        raise NotImplementedError(f'Unsupported Hypervisor: {hypervisor}')
+
+
+def test_guest_services_are_running(host):
+    """Check if guest agents services running and enabled"""
+    hyperv_services = ['hypervvssd', 'hypervkvpd', 'hypervfcopyd']
+    hypervisor = host.check_output("systemd-detect-virt")
+    if hypervisor == "kvm":
+        assert host.service("qemu-guest-agent.service").is_running
+        assert host.service("qemu-guest-agent.service").is_enabled
+    elif hypervisor == "oracle":
+        assert host.service("vboxadd-service.service").is_running
+        assert host.service("vboxadd-service.service").is_enabled
+    elif hypervisor == "vmware":
+        assert host.service("vmtoolsd.service").is_running
+        assert host.service("vmtoolsd.service").is_enabled
+    elif hypervisor == "microsoft":
+        for service in hyperv_services:
+            assert host.service(service).is_running
+            assert host.service(service).is_enabled
+    else:
+        raise NotImplementedError(f'Unsupported Hypervisor {hypervisor}')
+
+
+def test_insecure_vagrant_ssh_pub_key(host):
+    """Only Vagrant insecure public key should be present"""
+    authorized_keys = host.file("/home/vagrant/.ssh/authorized_keys").content_string
+    if len(authorized_keys.splitlines()) == 1:
+        # SHA256 checksum of "$vagrant_insecure_pub_key vagrant insecure public key"
+        assert host.file(
+            "/home/vagrant/.ssh/authorized_keys").sha256sum == "9aa9292172c915821e29bcbf5ff42d4940f59d6a148153c76ad638f5f4c6cd8b"
```

Comment:
Just question, auto test id keys?

Reply:
The vagrant box must only have the vagrant insecure SSH key's public part in its authorized_keys file.